test/e2e/scheduling/preemption.go

```diff
@@ -271,6 +272,150 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 
 		framework.ExpectEqual(podPreempted, true)
 	})
+
+	ginkgo.Context("PodTopologySpread Preemption", func() {
+		var nodeNames []string
+		var nodes []*v1.Node
+		topologyKey := "kubernetes.io/e2e-pts-preemption"
+		var fakeRes v1.ResourceName = "example.com/fakePTSRes"
+
+		ginkgo.BeforeEach(func() {
+			ginkgo.By("Trying to get 2 available nodes which can run pod")
+			nodeNames = Get2NodesThatCanRunPod(f)
+			ginkgo.By(fmt.Sprintf("Apply dedicated topologyKey %v for this test on the 2 nodes.", topologyKey))
+			for _, nodeName := range nodeNames {
+				framework.AddOrUpdateLabelOnNode(cs, nodeName, topologyKey, nodeName)
+
+				node, err := cs.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+				// update Node API object with a fake resource
+				nodeCopy := node.DeepCopy()
+				// force it to update
+				nodeCopy.ResourceVersion = "0"
+				ginkgo.By(fmt.Sprintf("Apply 10 fake resource to node %v.", node.Name))
+				nodeCopy.Status.Capacity[fakeRes] = resource.MustParse("10")
```

Comment:
is this reliable? will kubelet erase this since no such resource actually exist?

Reply:
For extended resources, no. We used to exercise this pattern for another test here:

https://github.com/kubernetes/kubernetes/blob/25651408aeadf38c3df7ea8c760e7519fd37d625/test/e2e/scheduling/preemption.go#L280